### PR TITLE
Allow larger EasyDMA buffer sizes on nrf52840

### LINKED ISSUE #17: Allow larger SPI buffer sizes on nrf52840 in nrf52-hal-common
the `maxcnt` register in the SPI hardware is 8 bits wide on the nrf52382 but wider on the 52840.  The boundary checking code currently constrains the maximum buffer size to 8 bit values which is safe but arbitrarily limiting on the 52840.

Come up with a reasonable way to express this range check that allows taking maximum advantage for the different target devices.

### PR DESCRIPTION
**Pull request by simonsso**
Expose const function in nrf52-hal-common which per platform
return limit of EasyDMA.

Closes #17

### CODE REVIEW

**Comment by Yatekii** [comment]
Why did you use a const fn instead of just a const variable?
Looks good to me on all other counts

**Comment by simonsso** [comment]
The _const fn_ evaluates to a constant at compile time, and I thought a get-value-function call looked clearer in the code, but it is a matter of taste.

The constant is also only in one place as it is tied to the EasyDMA block in the MCU more than to the individual function blocks.

**Comment by Yatekii** [comment]
> The const fn evaluates to a constant at compile time, and I thought a get-value-function call looked clearer in the code, but it is a matter of taste.

I am aware how it works, I just don't see the use of it. It's more confusing than not imo. Also it's not named get_ so it's not really a getter ;)

> The constant is also only in one place as it is tied to the EasyDMA block in the MCU more than to the individual function blocks.

How do you mean? I am not sure whether the MAXCNT is really an EasyDMA thing or a per-peripheral thing. But since the value is the same for UART/TWI/SPI, I am ok with having it the same constant for all 3 peripherals.

File: nrf52-hal-common/src/lib.rs [hunk 1]
```diff
@@ -31,6 +31,18 @@ pub mod prelude {
     pub use crate::uarte::UarteExt;
 }
 
+/// Length of Nordic EasyDMA differs for MCUs
+#[cfg(feature = "52832")]
+pub mod target_constants{
```
Comment (jamesmunns) [inline nrf52-hal-common/src/lib.rs]:
```suggestion
pub mod target_constants {
```

Formatting suggestion

File: nrf52-hal-common/src/lib.rs [hunk 2]
```diff
@@ -31,6 +31,18 @@ pub mod prelude {
...
+    // NRF52832 8 bits1..0xFF
+    pub const EASY_DMA_SIZE:usize = 255;
```
Comment (jamesmunns) [inline nrf52-hal-common/src/lib.rs]:
```suggestion
    pub const EASY_DMA_SIZE: usize = 255;
```

File: nrf52-hal-common/src/lib.rs [hunk 3]
```diff
@@ -31,6 +31,18 @@ pub mod prelude {
...
+}
+    // NRF52840 16 bits 1..0xFFFF
+#[cfg(feature = "52840")]
+pub mod target_constants{
```
Comment (jamesmunns) [inline nrf52-hal-common/src/lib.rs]:
```suggestion
pub mod target_constants {
```

File: nrf52-hal-common/src/lib.rs [hunk 4]
```diff
@@ -31,6 +31,18 @@ pub mod prelude {
...
+    pub const EASY_DMA_SIZE:usize = 65535;
```
Comment (jamesmunns) [inline nrf52-hal-common/src/lib.rs]:
```suggestion
    pub const EASY_DMA_SIZE: usize = 65535;
```

File: nrf52-hal-common/src/spim.rs [hunk 5]
```diff
@@ -226,16 +224,14 @@ impl<T> Spim<T> where T: SpimExt {
     /// This method uses the provided chip select pin to initiate the
     /// transaction, then transmits all bytes in `tx_buffer`.
     ///
-    /// The buffer must have a length of at most 255 bytes.
+
```
Comment (jamesmunns) [inline nrf52-hal-common/src/spim.rs]:
```suggestion
```

File: nrf52-hal-common/src/twim.rs [hunk 6]
```diff
@@ -263,13 +261,11 @@ impl<T> Twim<T> where T: TwimExt {
     )
         -> Result<(), Error>
     {
-        // This is overly restrictive. See:
-        // https://github.com/nrf-rs/nrf52-hal/issues/17
-        if wr_buffer.len() > u8::max_value() as usize {
+        if wr_buffer.len() > EASY_DMA_SIZE{
```
Comment (jamesmunns) [inline nrf52-hal-common/src/twim.rs]:
```suggestion
        if wr_buffer.len() > EASY_DMA_SIZE {
```

File: nrf52-hal-common/src/twim.rs [hunk 7]
```diff
@@ -263,13 +261,11 @@ impl<T> Twim<T> where T: TwimExt {
...
             return Err(Error::BufferTooLong);
         }
 
-        if rd_buffer.len() > u8::max_value() as usize {
+        if rd_buffer.len() > EASY_DMA_SIZE{
```
Comment (jamesmunns) [inline nrf52-hal-common/src/twim.rs]:
```suggestion
        if rd_buffer.len() > EASY_DMA_SIZE {
```

**Comment by jamesmunns** [comment]
Hey @simonsso, I made a couple formatting suggestions, I'm happy to merge once those are fixed!

**Comment by jamesmunns** [comment]
@simonsso looks good to me! Ready to merge, or do you have any other changes planned?

**Comment by simonsso** [comment]
Lets merge this first and I will move on to the dependent pull requests based on this.

**Pull request merged**